pkg/server/nodemanager/node_controller.go

```diff
@@ -269,6 +269,8 @@ func (c *NodeController) startSingleNode(node *api.Node, image cloud.Image, clou
 	)
 	if node.Spec.Spot {
 		instanceID, err = c.CloudClient.StartSpotNode(node, image, cloudInitData, c.Config.DefaultIAMPermissions)
+	} else if node.Spec.Dedicated {
```

Comment:
Here we could verify the `c.CloudClient` implements the `CloudClientDedicated` interface to verify this back-end supports dedicated hosts. If it doesn’t we issue a warning and fallback to `StartNode`.

I’m not sure if doing this now makes sense or not. Feel free to ignore this suggestion.